### ISSUE TITLE
Stock Entry Repack Issues- Scanned Batch No. Not Used in Serial and Batch Bundle

### ISSUE THREAD

**Issue by pmjd-code** [frappe/erpnext#36273]
### Information about bug

Using the Repack option in a Stock Entry to change one batch of an item into a new batch of another item. In v14 this works as expected, I can scan the Batch No, it populates the fields, I enter a new Item Code etc. and it is successfully Repacked to a new Item and new Batch No. However, it is not working the same in the Develop branch.

### Not Working - Using Barcode Scan Field to Enter Batch No
When the Batch No 000002 is scanned using the _Scan Barcode_ field on the DocType, 
![image](https://github.com/frappe/erpnext/assets/81377942/ce56f00b-96a6-456e-b124-cc646a977907)
it is correctly recognised and populates the Stock Entry Item field as expected. 
_Though why doesn't it know the Warehouse that the item is located in and populate it? Even if it was a Batch split in several locations, why does this not filter the Warehouse selection options?_
![Screenshot 2023-07-24 at 11-11-54 New Stock Entry - new-stock-entry-2](https://github.com/frappe/erpnext/assets/81377942/f5728e86-49fd-4ede-b40b-f7e81e7d657d)

A different warehouse, new item code, and quantity is entered as as required. The Repack is submitted, the original Batch No is correct,and no new Batch info is displayed but the option to view the Batch/Serial No are now visible after a refresh.
![Screenshot 2023-07-24 at 11-13-06 Repack - MAT-STE-2023-00026](https://github.com/frappe/erpnext/assets/81377942/bfc94ece-b842-4bb6-92e8-b67602752eb9)

However, on viewing the Batch No information, a different Batch with the same Item Code has been used in the Repack, not the Batch No selected.
![Screenshot 2023-07-24 at 11-13-39 Serial and Batch Summary](https://github.com/frappe/erpnext/assets/81377942/06fb7b23-a35e-44f3-ae0b-473c987403d6)

### Working With Issues - Entering Via Line Item

If I enter the Item Code directly in the line item, a more cumbersome way surely?, the dialog box appears to enter the Batch Nos, the scan of the Batch No is recognised correctly and the source warehouse again is not filtered and has to be selected.
![Screenshot 2023-07-24 at 11-37-04 New Stock Entry - new-stock-entry-1](https://github.com/frappe/erpnext/assets/81377942/88de07bc-0400-4693-9284-e9dc2649e953)

Once _Add Batch Nos_ is pressed the following warning appears, even though you didn't select the target warehouse and it shouldn't really be available as an option.
![Screenshot 2023-07-24 at 11-37-20 New Stock Entry - new-stock-entry-1](https://github.com/frappe/erpnext/assets/81377942/501f859a-ee7b-4b81-bbf6-cdb7c0b7bf5f)

The requisite information is entered for the Repack as before
 
![Screenshot 2023-07-24 at 11-38-11 Repack - MAT-STE-2023-00027](https://github.com/frappe/erpnext/assets/81377942/01b6a626-dc86-46a9-9e61-8edc72795d38)

But on Submit the following error appears
![Screenshot 2023-07-24 at 11-38-23 Repack - MAT-STE-2023-00027](https://github.com/frappe/erpnext/assets/81377942/b609f0ac-1e0c-4860-b8c3-b67ae41e7fbf)

After manually intervening and changing the Serial and Batch Bundle from _Inward_ to _Outward_, saving and not submitting the SBB, I can then successfully submit the Repack. On refresh there are no Batch details, as is the way of things now. On viewing the Batch Nos report, the correct item and Batch No has been successfully repacked.
![image](https://github.com/frappe/erpnext/assets/81377942/1e903d48-5939-41d6-bc0e-bf999a5559f2)

### Issues
- The original and easier option to enter only the Batch No to populate the Repack no longer functions correctly.
- The functional option has an issue with the transaction type (Inward/Outward)
- The functional option is more cumbersome for the end user to enter, i.e. scan/enter **Item Code** first, then enter Batch No and also know the correct Warehouse, rather than populating/filtering this information from the Batch No information.

### Module

stock

### Version

Frappe version - develop, updated 09:00 GMT 24 Jul 2023
ERPNext version - develop, updated 09:00 GMT 24 Jul 2023

### Installation method

manual install

**Comment by pmjd-code** [comment]
@s-aga-r is this an issue with the Serial and Batch Bundle implementation?

**Comment by pmjd-code** [comment]
Just updated to latest development branch and tried again, issue with Inward/Outward seems to have been resolved.

**Comment by s-aga-r** [comment]
> @s-aga-r is this an issue with the Serial and Batch Bundle implementation?

Serial and Batch Bundle is added in v15 (develop).

**Comment by pmjd-code** [comment]
> Serial and Batch Bundle is added in v15 (develop).

Yes and that is the version I'm reporting the error in, please let me know if anything in my report is unclear or ambiguous.
